tests/jballerina-unit-test/src/test/resources/test-src/narrowing/assignment_to_narrowed_vars_in_loops_negative.bal

```diff
@@ -679,3 +679,42 @@ function f23() {
         boolean _ = m is error;
     }
 }
+
+function testInLoopAssignmentsWithQueryAction() returns error? {
+    (int|string)[] x = [1, 2];
+    int|string? p = 10;
+
+    if p is int|string {
+        check from var a in x
+            where p is int
+            do {
+                p = 12;
```

Comment:
error msg here "invalid attempt to assign a value to a variable narrowed outside the query action" is not correct; isn't it?

Reply:
Check other places as well

Reply:
`p` is defined outside the query and it's narrowed outside the query. So it should right?

Reply:
my bad. this is correct